integration-tests/src/test/java/com/linkedin/kafka/clients/producer/LiKafkaProducerIntegrationTest.java

```diff
@@ -195,5 +209,77 @@ public void close() {
     assertEquals(1, messageCount);
   }
 
+  @Test
+  public void testCustomPartitioningOfLargeMessage() throws Exception {
+    Random random = new Random();
+    int numPartitions = 2 + random.nextInt(19); // [2-20]
+    String topic = "testCustomPartitioner-" + UUID.randomUUID();
+
+    AdminClient adminClient = createRawAdminClient(null);
+    CreateTopicsResult createTopicResult = adminClient.createTopics(Collections.singletonList(new NewTopic(topic, numPartitions, (short) 1)));
+    createTopicResult.all().get(1, TimeUnit.MINUTES);
+    TopicDescription topicDescription = adminClient.describeTopics(Collections.singletonList(topic)).all().get(1, TimeUnit.MINUTES).get(topic);
+    Assert.assertEquals(topicDescription.partitions().size(), numPartitions, "expected topic to have "
+        + numPartitions + " partitions. instead get " + topicDescription.partitions());
+
+    Properties producerPropertyOverrides = new Properties();
+    producerPropertyOverrides.put(ProducerConfig.PARTITIONER_CLASS_CONFIG, RoundRobinPartitioner.class.getCanonicalName());
```

Comment:
Since this partitioner starts at zero check that all the message segments end up on partition zero.